### PR TITLE
Revise docstring of ApproximateNearestNeighbors to be more accurate

### PR DESCRIPTION
**Pull request by lijinf2**
1. Added 'cosine' option for the 'metric' params
2. Added the supported metrics for individual algorithm ('ivfflat', 'ivfpq', 'cagra')
3. Explained spark_rapids_ml leverages cuvs for computation and follows the APIs of cuml.

### CODE REVIEW

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/knn.py [hunk 1]
```diff
@@ -966,7 +967,10 @@ class ApproximateNearestNeighbors(
             Note cuml requires M * n_bits to be multiple of 8 for the best efficiency.
 
     metric: str (default = "euclidean")
-        the distance metric to use. 'ivfflat' algorithm supports ['euclidean', 'sqeuclidean', 'l2', 'inner_product'].
+        the distance metric to use with the default set to "euclidean" (following cuml conventions, though cuvs defaults to "sqeuclidean").
+        The 'ivfflat' and 'ivfpq' algorithms support ['euclidean', 'sqeuclidean', 'l2', 'inner_product', 'cosine'].
+        The 'cagra' algorithm supports ['sqeuclidean'], and when using 'cagra' as an algorithm,
+        the metric must be explicitly set to 'sqeuclidean'.
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/knn.py:973]: Do we have a helpful error message if not the case?

Comment (lijinf2) [reply python/src/spark_rapids_ml/knn.py:973]: Added the error message and a test case.

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**